File: Day259_2025/Q259_GFG.java
import java.util.*;

class Q259_GFG {
    public int evaluatePostfix(String[] arr) {
        Stack<Integer> st = new Stack<>();
        
        for (String token : arr) {
            if (token.equals("+") || token.equals("-") || token.equals("*") 
                || token.equals("/") || token.equals("^")) {
                
                int op2 = st.pop();
                int op1 = st.pop();
                int res = 0;
                
                switch (token) {
                    case "+": res = op1 + op2; break;
                    case "-": res = op1 - op2; break;
                    case "*": res = op1 * op2; break;
                    case "/": res = op1 / op2; break;  // Integer division
                    case "^": res = (int) Math.pow(op1, op2); break;
                }
                
                st.push(res);
            } else {
                st.push(Integer.parseInt(token));
            }
        }
        
        return st.peek();
    }
}

File: Day259_2025/Q259_GFG.py
class Solution:
    def evaluatePostfix(self, arr):
        st = []
        
        for token in arr:
            if token in "+-*/^":
                op2 = st.pop()
                op1 = st.pop()
                
                if token == '+':
                    res = op1 + op2
                elif token == '-':
                    res = op1 - op2
                elif token == '*':
                    res = op1 * op2
                elif token == '/':
                    res = op1 // op2  # Integer division
                elif token == '^':
                    res = op1 ** op2
                
                st.append(res)
            else:
                st.append(int(token))
        
        return st[-1]
